vllm/block.py

```diff
@@ -37,5 +37,43 @@ def __repr__(self) -> str:
                 f'computed={self.computed})')
 
 
-# Mapping: logical block number -> physical block.
-BlockTable = List[PhysicalTokenBlock]
+class BlockTable:
+
+    def __init__(self, blocks=None):
+        self._blocks: List[PhysicalTokenBlock] = []
+        self._block_ids: List[int] = []
+
+        if blocks is not None:
+            for block in blocks:
+                self.append(block)
+
+    def append(self, block: PhysicalTokenBlock):
+        self._blocks.append(block)
+        self._block_ids.append(block.block_number)
+
+    def __len__(self) -> int:
+        return len(self._blocks)
+
+    def __getitem__(self, key):
+        return self._blocks[key]
+
+    def __setitem__(self, key, block):
+        if isinstance(key, slice):
+            self._blocks[key] = block
+            self._block_ids[key] = [b.block_number for b in block]
+        else:
+            self._blocks[key] = block
+            self._block_ids[key] = block.block_number
+
+    def reset(self):
+        self._blocks = []
+        self._block_ids = []
+
+    def copy(self) -> "BlockTable":
+        return BlockTable(self._blocks)
```

Comment:
is it a `fork` or copy? do we need to copy the last block if it is not full?

Reply:
This is a full copy of the object. Fork() and similar methods use it internally (in block_manager_v1 it was used to copy the list of physical blocks)